src/uwtools/drivers/forecast.py

```diff
@@ -0,0 +1,237 @@
+        help='If provided, validate configuration but do not run the forecast',
+    )
+    parser.add_argument(
+        '-v', '--verbose',
+        action='store_true',
+        help='If provided, print all logging messages.',
+        )
+    parser.add_argument(
+        '-q', '--quiet',
+        action='store_true',
+        help='If provided, print no logging messages',
+        )
+    parser.add_argument(
+        '-l', '--log_file',
+        help='Optional path to a specified log file',
+        default=os.path.join(os.path.dirname(__file__), "forecast.log")
+        )
+    return parser.parse_args(argv)
+
+class Forecast:
+
+    ''' 
+    This base class provides the interface to methods used to read in
+    a user-provided YAML configuration file, convert it to the required
+    config file, fix files, and namelist for a forecast. Subsequent
+    methods will be used to stage the input files and run the forecast.        
+    
+    Attributes
+    ----------
+    config_path : Path
+        The file path to the configuration file to be parsed.
+
+    Methods
+    -------
+    requirements()  
+        Recursively parse config and platform files to determine and
+        fill in any dependencies.
+
+    resources()
+        Determine necessary task objects and fill in resource requirements of each.
+        Returns a Config object containing the HPC resources needed.
+
+    validate()
+        Validates the objects generated by the driver from the provided
+        config and platform files. If the --dry_run flag is provided, complete all
+        stages through validation and print results without running the forecast.
+
+    create_model_config()
+        Collects all the user inputs required to create a model config
+        file, calling the existing model config tools. 
+
+    stage_fix_files()   
+        Holds the knowledge for how to modify a list of fix files and
+        stages them in the working directory.
+
+    create_namelist()
+        Collects all the user inputs required to create a namelist
+        file, calling the existing namelist config tools.
+
+    stage_output_files()
+        Holds the knowledge for how to modify a list of output files and
+        stages them in the working directory.
+
+    job_card()
+        Turns the resources config object into a batch card for the 
+        configured Task. Can be called from the command line on the front end 
+        node to show the user what the job card would have looked like.
+
+    run()
+        Runs the forecast executable with the namelist file and staged
+        input files. This will both build the executable and run it.
+    
+    '''
+
+    def __init__(self, argv):
+
+        '''
+        Initialize the Forecast driver.
+
+        '''
+        # Parse user arguments
+        user_args = parse_args(argv)
+
+        # Set up logging
+        name = f"{inspect.stack()[0][3]}"
+        log = cli_helpers.setup_logging(user_args, log_name=name)
```

Comment:
So far, we've been creating our logger in the caller, and getting it in the init method of the class that uses it.

Reply:
OHHH wait, yes, my mistake, as its in the base class but its not abstract, so actual creation should go to the concrete class, right?

Reply:
Moved logging to main()